test/components/groupTests.ts

```diff
@@ -1,258 +1,268 @@
-    let t1 = svg.select(".test-box1");
-    let t2 = svg.select(".test-box2");
-    let t3 = svg.select(".test-box3");
-    TestMethods.assertWidthHeight(t1, 10, 10, "rect1 sized correctly");
-    TestMethods.assertWidthHeight(t2, 400, 400, "rect2 sized correctly");
-    TestMethods.assertWidthHeight(t3, 400, 400, "rect3 sized correctly");
-    svg.remove();
-  });
+      svg.remove();
+    });
 
-  it("detach()", () => {
-    let c1 = new Plottable.Component().addClass("component-1");
-    let c2 = new Plottable.Component().addClass("component-2");
-    let cg = new Plottable.Components.Group([c1, c2]);
+    it("can move Components to other Groups after anchoring", () => {
+      let svg = TestMethods.generateSVG();
 
-    let svg = TestMethods.generateSVG(200, 200);
-    cg.renderTo(svg);
+      let group1 = new Plottable.Components.Group();
+      let group2 = new Plottable.Components.Group();
+      let component = new Plottable.Component();
 
-    let c1Node = svg.select(".component-1").node();
-    let c2Node = svg.select(".component-2").node();
+      group1.append(component);
 
-    assert.isNotNull(c1Node, "component 1 was added to the DOM");
-    assert.isNotNull(c2Node, "component 2 was added to the DOM");
+      group1.renderTo(svg);
+      group2.renderTo(svg);
 
-    c2.detach();
+      assert.strictEqual(group2.components().length, 0, "second group should have no component before movement");
 
-    c1Node = svg.select(".component-1").node();
-    c2Node = svg.select(".comopnent-2").node();
+      assert.strictEqual(group1.components().length, 1, "first group should have 1 component before movement");
 
-    assert.isNotNull(c1Node, "component 1 is still in the DOM");
-    assert.isNull(c2Node, "component 2 was removed from the DOM");
+      assert.strictEqual(component.parent(), group1, "component's parent before moving should be the group 1");
 
-    cg.detach();
-    let cgNode = svg.select(".component-group").node();
-    c1Node = svg.select(".component-1").node();
+      assert.doesNotThrow(() => group2.append(component), Error, "should be able to move components between groups after anchoring");
 
-    assert.isNull(cgNode, "component group was removed from the DOM");
-    assert.isNull(c1Node, "componet 1 was also removed from the DOM");
+      assert.strictEqual(group2.components().length, 1, "second group should have 1 component after movement");
 
-    cg.renderTo(svg);
-    cgNode = svg.select(".component-group").node();
-    c1Node = svg.select(".component-1").node();
+      assert.strictEqual(group1.components().length, 0, "first group should have no components after movement");
 
-    assert.isNotNull(cgNode, "component group was added back to the DOM");
-    assert.isNotNull(c1Node, "componet 1 was also added back to the DOM");
+      assert.strictEqual(component.parent(), group2, "component's parent after movement should be the group 2");
 
-    svg.remove();
-  });
+      svg.remove();
```

Comment:
for lines 149 - 163, let's
- Either remove all \n between asserts or group the first 3 and the last 3 together
- Assert for group 1 first, then for group 2 (swap lines 151-153 and 157-159)